crates/ruff_python_ast/src/node.rs

```diff
@@ -3139,7 +3139,7 @@ impl AstNode for ast::PatternMatchSingleton {
         V: PreorderVisitor<'a> + ?Sized,
     {
         let ast::PatternMatchSingleton { value, range: _ } = self;
-        visitor.visit_constant(value);
+        visitor.visit_singleton(value);
```

Comment:
What about constant's in other positions? e.g. do we call this function when visiting a `ExprConstant` too?

Reply:
Yes, we do:

https://github.com/astral-sh/ruff/blob/db9d767841f8e5a1ec49275e6731d7c21777c256/crates/ruff_python_ast/src/node.rs#L2681-L2687

I'm not sure if there should be a `visit_singleton` or not. Do we define visitor methods for the enum values as well? Wouldn't they be accounted for when visiting the enclosing node? Looking at `Constant`, we do.

Reply:
I'm not sure. I guess it can be useful when the same non-nodes may have different parent nodes but I would be okay only having visitors for nodes. @charliermarsh should have more context on why `visit_constant` exists.